### DISCUSSION TITLE
Bringing back vite_hanami

### DISCUSSION THREAD

**Discussion by krzykamil** [ElMassimo/vite_ruby#445] [General]
Hey

So currently the vite_hanami is broken to my understanding. It is probably meant for hanami pre 2.0, and now we have 2.1 which has its own, default way of doing views and assets.

However if someone wants to use hanami views, but not assets and wants to use vite, the current state of this gem, does not work for this purpose.

The tag helpers use non-existing methods, need to be included in a different way, in a different place, the vite-dev folder in public does not get generated etc.

In general for hanami, the gem is severely out-of-date. I was wondering if there are any plans/work in progress, to get it working again for hanami 2.1? If not, would you accept PR's that try to make it work with the newer hanami? It would be aimed for integrations that use hanami-views but skip hanami-assets I guess, although I am not sure. Without hanami-assets there are no tag halpers like javascript_tag etc. but everything else would be obsolete from that gem if using vite.

**Comment by ElMassimo** [comment]
Hi Krzysztof!

The idea was for `vite_hanami` to eventually support Hanami 2 (it wasn't released back then).
I don't use Hanami myself, and no users have requested this, so I assume folks using Hanami are typically not interested in a bundler like Vite.

To avoid having to manage tag helpers such as `javascript_include_tag`, I'd like for `vite_hanami` to be built on top of `hanami-assets` or whatever module is in charge of providing those.

**Comment by krzykamil** [comment]
Thanks for the quick response!

In that case I would like to fork the repo and give it a shot to upgrade the current implementation into hanami 2.1 using `hanami-assets` . I will post a PR if I succeed up for review.

Also thanks for the gem overall. I've been using the rails version and it really is nice and convenient :)

**Reply by JoeeGrigg** [reply]
Just wondering how you got on with this? I'm new to both Vite and Hanami so feel completely unqualified to try and get everything plugged together but it would be a great combination if possible 😄

**Reply by krzykamil** [reply]
Right now, the way hanami-assets builds its paths of assets and lookups does not allow to use its asset helpers to point to vite assets. It would either require monkey-patching or using vite separate of hanami-assets instead of on top of it, which was the idea.
I was thinking about trying to adapt hanami assets library to allow for handling vite builds etc. But that is a substantially bigger workload, and I haven't gotten around to it yet.
